Kickstarter-iOS/Features/PledgePaymentMethods/Controller/PledgePaymentMethodsViewController.swift

```diff
@@ -144,33 +146,48 @@ final class PledgePaymentMethodsViewController: UIViewController {
   // MARK: - Functions
 
   private func goToPaymentSheet(data: PaymentSheetSetupData) {
-    PaymentSheet.FlowController
-      .create(
-        setupIntentClientSecret: data.clientSecret,
-        configuration: data.configuration
-      ) { [weak self] result in
-        guard let strongSelf = self else { return }
+    let completion: (Result<PaymentSheet.FlowController, Error>) -> Void = { [weak self] result in
+      guard let strongSelf = self else { return }
 
-        switch result {
-        case let .failure(error):
-          strongSelf.viewModel.inputs.shouldCancelPaymentSheetAppearance(state: true)
-          strongSelf.messageDisplayingDelegate?
-            .pledgeViewController(strongSelf, didErrorWith: error.localizedDescription)
-        case let .success(paymentSheetFlowController):
-          let topViewController = strongSelf.navigationController?.topViewController
-          let paymentSheetShownWithinPledgeContext = topViewController is PledgeViewController
-
-          if paymentSheetShownWithinPledgeContext {
-            strongSelf.paymentSheetFlowController = paymentSheetFlowController
-            strongSelf.paymentSheetFlowController?.presentPaymentOptions(from: strongSelf) { [weak self] in
-              guard let strongSelf = self else { return }
-
-              strongSelf.confirmPaymentResult(with: data.clientSecret)
-            }
-          }
-          strongSelf.viewModel.inputs.stripePaymentSheetDidAppear()
+      switch result {
+      case let .failure(error):
+        strongSelf.viewModel.inputs.shouldCancelPaymentSheetAppearance(state: true)
+        strongSelf.messageDisplayingDelegate?
+          .pledgeViewController(strongSelf, didErrorWith: error.localizedDescription)
+      case let .success(paymentSheetFlowController):
+        let topViewController = strongSelf.navigationController?.topViewController
+
+        assert(
```

Comment:
Note that I got rid of the `if paymentSheetShownWithinPledgeContext` and changed it to an assert, instead. AFAIK this view controller is already only ever used from a pledge context.